src/main/java/cz/cvut/kbss/analysis/service/FailureModesTableRepositoryService.java

```diff
@@ -154,39 +154,37 @@ private List<Map<String, Object>> computeTableRows(FailureModesTable table, List
                         .stream()
```

Comment:
line #119 - question for optimization - this line will search for a path between the root and the leaf in the tree. Isn't the path already found and stored in the local variable `FailureModesRow r`? Or maybe this is necessary because the path might have changed?